### DISCUSSION TITLE
How to use stamina with class method?

### DISCUSSION THREAD

**Discussion by triumph1** [hynek/stamina#41] [Q&A] [answered]
Hi! Thanks for your amazing work
(I have always wrote retry on my own, now you made my life easy!)

```python

import asyncio
from datetime import timedelta

import httpx
import stamina


class GoogleClient:

    @classmethod
    @stamina.retry(
        on=httpx.HTTPError, attempts=3, timeout=timedelta(seconds=10)
    )
    async def get_google(cls):
        async with httpx.AsyncClient() as client:
            response = await client.get("https://www.google.com")
        return response

result = asyncio.run(GoogleClient.get_google())
print(result.content)

```

this works well, is this a right way to use stamina with @classmethod?

**Comment by hynek** [comment, marked as answer]
Is there a reason for you to think it's not? Stamina shouldn't care about the type of the type of the callable it's wrapping so I think so? If you run into any problems or interesting effects, please do let me know.

**Reply by triumph1** [reply]
Thanks! I just wanted to make sure. works like charm so far :)
